packages/graphql/src/translate/translate-read.ts

```diff
@@ -124,30 +126,77 @@ function translateRead({ node, context }: { context: Context; node: Node }): [st
         }
 
         if (optionsInput.sort && optionsInput.sort.length) {
+            if (hasLimit) {
+                const sortFields = optionsInput.sort.map((sort) => Object.keys(sort)[0]);
+                const orderByProjection = createProjectionAndParams({
+                    node,
+                    context,
+                    fieldsByTypeName: {
+                        [node.name]: Object.fromEntries(
+                            Object.entries(fieldsByTypeName[node.name]).filter(([key]) => sortFields.includes(key))
+                        ),
+                    },
+                    varName,
+                });
+                [limitOrderProjStr] = orderByProjection;
+                cypherParams = { ...cypherParams, ...orderByProjection[1] };
+            }
+
             const sortArr = optionsInput.sort.reduce((res: string[], sort: GraphQLSortArg) => {
                 return [
                     ...res,
                     ...Object.entries(sort).map(([field, direction]) => {
-                        return `${varName}.${field} ${direction}`;
+                        if (!cypherSort && node.cypherFields.some((f) => f.fieldName === field)) {
+                            cypherSort = true;
+                        }
+                        return `${hasLimit ? limitOrderVarName : varName}.${field} ${direction}`;
                     }),
                 ];
             }, []);
 
-            sortStr = `ORDER BY ${sortArr.join(", ")}`;
+            sortStr = sortArr.length ? `ORDER BY ${sortArr.join(", ")}` : "";
         }
     }
 
-    const cypher = [
-        matchAndWhereStr,
-        authStr,
-        ...(projAuth ? [`WITH ${varName}`, projAuth] : []),
-        ...connectionStrs,
-        ...interfaceStrs,
-        `RETURN ${varName} ${projStr} as ${varName}`,
-        ...(sortStr ? [sortStr] : []),
-        offsetStr,
-        limitStr,
-    ];
+    let cypher: string[] = [];
+
+    if (!hasLimit) {
+        cypher = [
+            matchAndWhereStr,
+            authStr,
+            ...(sortStr && !cypherSort ? [`WITH ${varName}`, sortStr] : []),
+            ...(projAuth ? [`WITH ${varName}`, projAuth] : []),
+            ...connectionStrs,
+            ...interfaceStrs,
+            `RETURN ${varName} ${projStr} as ${varName}`,
+            ...(sortStr && cypherSort ? [sortStr] : []),
+            offsetStr,
+        ];
+    } else {
+        cypher = [
+            "CALL {",
+            matchAndWhereStr,
+            authStr,
+            ...(sortStr && !cypherSort
```

Comment:
The `limitOrderProjStr` handles a projection including cypher fields, and so can be sorted here.

The downside of this is that the cypher projection would potentially be evaluated twice, once in the CALL block, and once outside.

It may be possible to optimize this by projecting fields participating in the sort inside the `CALL` block, and those which don't outside, then merging the two projections into one final projection which is returned.  I don't know if Cypher supports something like spreading.

`RETURN this { **this_sort, foo: ... } AS this`